cpp/bridges_module/algorithm/bridges.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <mg_graph.hpp>
+
+namespace bridges_util {
+
+/// Simple struct that keeps the state of nodes in algorithms that rely on
+/// DFS traversal.
+struct NodeState {
+  std::vector<bool> visited;
+  std::vector<std::uint64_t> discovery, low_link, depth;
+  std::vector<std::uint64_t> parent;
+  std::uint64_t counter;
+
+  explicit NodeState(std::uint64_t number_of_nodes) {
+    visited.resize(number_of_nodes, false);
+    discovery.resize(number_of_nodes, 0);
+    low_link.resize(number_of_nodes, 0);
+    counter = 0;
+  }
+
+  void Update(std::uint64_t node_id) {
+    counter++;
+    visited[node_id] = true;
+    discovery[node_id] = counter;
+    low_link[node_id] = counter;
+  }
+};
+
+void BridgeDfs(uint64_t node_id, uint64_t parent_id, bridges_util::NodeState *state,
+               std::vector<mg_graph::Edge<>> *bridges, const mg_graph::GraphView<> &graph);
+}  // namespace bridges_util
```

Comment:
What's the different in the DFS implementation between this one and the Bcc?

Reply:
There is some kind of difference in the internal logic, I might want to check on this one further because a lot of algorithms are going to use DFS. Might be useful if DFS only employs basic logic in mg_graph, but then uses the `consumer` pattern to do custom stuff that each different algorithm requires.

Reply:
Yep, I was thinking about something like that. But, if the common part is mixed with custom parts you will end up with a mess, and repeating the DFS logic for each algorithm would be better.

Reply:
I think the common part largely intersects with the custom part, which would be difficult to implement as a utility.